### ISSUE TITLE
adding OData to project not working

### ISSUE THREAD

**Issue by figuerres** [rbeauchamp/Swashbuckle.OData#87]
I am getting started with Swashbucle and this OData module.
I got the basic Swashbucle / swagger page to work.
I added the OData module and tried to add the basic config line as the web pages show.
I get the page where it's trying to build the documation page and it just hangs there with no change.

if I comment out the line
 c.CustomProvider(defaultProvider => new ODataSwaggerProvider(defaultProvider, c));
then I get the swagger for the web api methods.

trying to figure out what is wrong but I do not have a clue where to look just yet.

**Comment by figuerres** [comment]
Some more information that may help in my problem:
I created a small project and I have the swagger page coming up and showing that it know I have an OData endpoint.

My main project has about 15 controllers right now and is not using restier.
each controller uses a generic base class to handle some common code for each table that is mapped to an odata controller.

I wonder if the generic base class is creating an issue with the processing to publish the swagger ?
I say this as in trying to debug I see that the call into adding the OData provider is going into some kind of LINQ IEnumerable code in .Net and I wonder if it's getting hung up in trying to work out my base class and how it provides the data.
I have not used RESTier and will see if that might help me with the structure of my api and also get this to work.

**Comment by rbeauchamp-lendingtree** [comment]
Hey @figuerres Is there some sample code or a repo you can point us to that reproduces the issue? Thanks!

**Comment by figuerres** [comment]
Richard, it took some time to get to the bottom but now I know what is happening and I am looking for good long term fix.
A backstory: I work for a company that needs to replace an old line of business web application.
The CIO wants to use angular and OData. 
I was brought on a few months back and have started a new database design and creating OData controllers for it based on the default scaffolding.
A while back I created a generic base class to make standing up new tables a simple task. So far the prototype works, but I have seen that each contoller class is referencing the tables in the Jason meta data, but I thought that is needed to allow the OData server framework to be able to query and expand entity collections as needed.

So I have about 30 tables and 30 controllers and each contoller has its get, post, patch and delete verbs

So the swashbuckle OData code starts to iterate over this to generate documentation....

And it looked like the web app hung...

I now see that it was building the html for. 30 * 30 * 4 items and all the metadata for that.
Huge html file that takes about 5 minutes to generate and download.

So I am now looking at how to better create the controller's to be flexible but not repeat all that stuff.

I tried to hook up wrestler but that seems to not work and gives me a 404 and I have not figured out why.
And I do not know if it will be a better way to do this.
We have a standard set of items in each table that right now my generic base class handles.
I do not know if I can maker resttier do this.
Ok that was quite a story there.... now you know.

**Comment by rbeauchamp** [comment]
@figuerres Actually sounds like a fun project! Hey, can you attach or show your SwaggerConfig? Specifically all of the `c.CustomProvider(.... `   lines. By default Swashbuckle.OData should not be including the "navigation properties", which is what I think you are referring to when you say "30 tables".

**Comment by figuerres** [comment]
the issue is in learning how to setup OData for a larger project I saw that the default scaffolding was making multiple objects with
Builder.EntitySet<ComponentRoleProperty>("componentroleproperties");
var model =   Builder.GetEdmModel();

and as I have many relations I got tiered of having so many declarations so I made one and then  reused it  like this
    config.MapODataServiceRoute("atrackapiAudits", "atrakapi/Audits", model);
            config.MapODataServiceRoute("atrackapiNotes"         , "atrakapi/Notes", model);
            config.MapODataServiceRoute("atrackapiAdresses"      , "atrakapi/Addresses", model);

so then every controller sees all the tables and can expand on any that are related.
that worked ok until I then tried to add the swagger.

I am trying out Restier to see if that will work for me as it handles the model more like what I tried to do.
I now have to see if restier will allow me to use the pattern I have  where each table has a set of columns that need to be updated.
I have a set of date and ID  columns to track creation and change of a record and a logical delete flag that I use in my "standard pattern"
so in my data model I have an Interface that all my entity tables implement to standardize that part of the model.
if I can make my pattern work with the restier bits then I am going to be in good shape... I think.

**Comment by figuerres** [comment]
Well just a last comment here, I did not understand the routing and controller setup very well at first and that was leading to my creating a *LOT* of redundant routes that led to the mess I had!
get rid of the rout statements and now things are looking 100% better.

RESTier : interesting package but I think they need to finish it. I tried to work with it a bit but in the end I feel that my list of controllers is "Good Enough" as it works and is not too complicated.
